dracut/modules.d/99kiwi-lib/kiwi-lib.sh

```diff
@@ -41,8 +41,7 @@ function lookup_disk_device_from_root {
         root_device=/dev/$(basename "$(readlink "${root_device}")")
     fi
     for disk_device in $(
-        lsblk -p -n -r -s -o NAME,TYPE "${root_device}" |\
-            grep disk | cut -f1 -d ' '
+	lsblk -p -n -f -o PKNAME "${root_device}"
```

Comment:
Thanks for your review.
If I understand it correctly, you pass a root_device (like /dev/sda3 or /dev/md126p3) and try to find the parent of this (/dev/sda or /dev/md126).:

For a non-raid and a non-fake-raid PC I'll get with `` :
```
# lsblk -p -n -r -s -o NAME,TYPE /dev/sda3
/dev/sda3 part
/dev/sda disk
```
, where I get `/dev/sda` line by the ` grep disk | cut -f1 -d ' '` command.


For a fake-raid I'll get from
```
# lsblk -p -n -r -s -o NAME,TYPE /dev/md126p3
/dev/md126p3 md
/dev/md126 raid1
/dev/sdb disk
/dev/sda disk
/dev/sdb disk
/dev/sda disk
```
Here, I'll get `/dev/sda` and `/dev/sdb` as results. This will break the disk, if I would start to work with sda or sdb instead of md126.

When I'm using the other command on a non-raid:
```
# lsblk -p -n -f -o PKNAME /dev/sda3
/dev/sda
```
and on the raid 
```
# lsblk -p -n -f -o PKNAME /dev/md126
/dev/md126
```

But I see: If you pass /dev/sda or /dev/md126 to this function to get /dev/sda or /dev/md126 back, this method creates multiple outputs. I think, a solution would be to limit the output to one line like:
```shell
lsblk -p -n -f -o PKNAME "${root_device}" | cut -n1
```

Sincerly, the more obvious change like:
```shell
lsblk -p -n -r -s -o NAME,TYPE "${root_device}" |\
grep -E "disk|raid1" | cut -f1 -d ' '
```
does not work on md, because it will also find the underlying hard drives.

What do you think?

Reply:
The lsblk call should be changed from

```
lsblk -p -n -r -s -o NAME,TYPE "${root_device}" |\
            grep disk | cut -f1 -d ' '
```

to

```
lsblk -p -n -r -s -o NAME,TYPE "${root_device}" |\
            grep -E "disk|raid" | cut -f1 -d ' '
```

Similar to what I had done in #1185

Reply:
Yes, this would be the obvious solution. But the command gives:
```shell
# lsblk -p -n -r -s -o NAME,TYPE /dev/md126p3
/dev/md126p3 md
/dev/md126 raid1
/dev/sdb disk
/dev/sda disk
/dev/sdb disk
/dev/sda disk
```
Filtering disk **or** raid will also give the unwanted `/dev/sda` and `/dev/sdb`
```shell
# lsblk -p -n -r -s -o NAME,TYPE /dev/md126p3 | grep -E "disk|raid"
/dev/md126 raid1
/dev/sdb disk
/dev/sda disk
/dev/sdb disk
/dev/sda disk
```

Reply:
yes what I wrote would be the first part of the solution. The other part would be to identify which storage devices belongs to the raid and consolidate the list. In the code you have changed you see that I did something similar if the root device is on multipath. Unfortunately I don't have access to a fake raid system but I would check if mdadm or dmraid can be used to identify which raw disk storage devices belongs to the raid.

That code part is still missing